### ISSUE TITLE
XHR errors 

### ISSUE THREAD

**Issue by DanielCryptoK** [swissspidy/web-stories-wp#11533]
I have countless XHR errors,  click link https://search.google.com/test/amp/result?id=AWPDp0iQAUxVRhwEXOk5iw and then view tested page to see javascript console messages. Could this error be the thing that is stopping my voice over from working correctly, see this url for what I mean https://bitcoin-casino-no-deposit-bonus.com/web-stories/bitstarz/

For my second website I am having different javascript console errors as seems to be a problem with loading even though it does not look that way. https://search.google.com/test/amp/result?id=gp6IKlzKpJjAygkRZp0KKQ

**Comment by DanielCryptoK** [comment]
I think I may have worked out what is causing this bug as I have just replied to a thread about the audio captions. The error is related to VTT files as it is only showing on the web stories I have added these. How to fix?

**Comment by swissspidy** [comment]
Do these errors happen to you also when you visit the story manually in incognito mode? Can you please test this?

In my quick testing there were no such errors in incognito mode _or_ when deleting the service worker.

If that is also the case for you, you might want to file a bug at https://github.com/ampproject/amphtml mentioning this potential incompatibility.

PS. Please use the support forums next time for support questions like this, since we do not provide support via GitHub. I have mentioned this to you multiple times so far.

**Comment by DanielCryptoK** [comment]
Hi,

I would love to ask in the wordpress support but they gave me a lifetime ban because I included by mistake a url in a line of code that was broken after been warned about not posting urls which is crazy when it is code with a url in it. I have asked for their legal department so that I can try to overturn the ban but they have not got back to me. It means that the moderators have cut my business off from getting vital support for plugins that power my businesses. Are you able to get me back reinstalled as it is such a shame to see cancel culture has taken over the wordpress forum. 

If you are able to provide me access to a google support forum for web stories that is not linked to wordpress forums I could try that to appease you.

I have done a test in incognito and the javascript errors are still showing in the amp test. I do not understand how it is possible that we are both seeing different results???

What do you mean by deleting the service worker?

You did the test for this url? https://bitcoin-casino-no-deposit-bonus.com/web-stories/bitstarz/

![Web capture_23-5-2022_172952_search google com](https://user-images.githubusercontent.com/45933242/169920399-dabd7a10-bda4-4d17-acfb-673bf66a78e1.jpeg)

**Comment by swissspidy** [comment]
Unfortunately we are not able to help you with banning issues on the WordPress.org forums. We also do not offer/maintain our own support forums.

I am not talking about testing in the AMP Test Tool, but by opening the story directly. Let me rephrase. Follow these steps:

1. Open https://bitcoin-casino-no-deposit-bonus.com/web-stories/bitstarz/ directly in your browser and click through the story pages
2. Check whether video/audio is playing normally and whether there are any JavaScript console messages. _Press F12 to open the browser console and see these messages_
3. There might be errors like this, matching what is reported by the AMP Test tool:  
    <img width="1166" alt="Screenshot 2022-05-24 at 09 49 45" src="https://user-images.githubusercontent.com/841956/169978634-29c16c6d-47da-477e-a029-65d1574dad75.png">

Now repeat this, _but in an incognito window_.

1. Open https://bitcoin-casino-no-deposit-bonus.com/web-stories/bitstarz/ directly in your browser _in incognito mode_ and click through the story pages
2. Check whether video/audio is playing normally and whether there are any JavaScript console messages.
3. There might be _no errors anymore_

If there are indeed  no errors anymore in incognito, try this _in a normal browser window_ again:

1. Open https://bitcoin-casino-no-deposit-bonus.com/web-stories/bitstarz/ directly in your browser
2. Press F12 to open dev tools, click on "Application", "Service Workers", and then "Unregister":    
   <img width="1116" alt="Screenshot 2022-05-24 at 09 48 10" src="https://user-images.githubusercontent.com/841956/169978984-3cc96511-7516-4233-ac01-4f3ba1b85c6d.png">
3. Reload the story and click through the story pages
4. Check whether video/audio is playing normally and whether there are any JavaScript console messages.
3. There might be _no errors anymore_

**Comment by swissspidy** [comment]
No. That's not what I am saying. Also, https://github.com/ampproject/amphtml is not responsible for the AMP Test tool.

What I am saying is:

Can you please follow the steps I shared above and confirm whether you still errors when you follow these steps.

Then we can go from there.

**Comment by DanielCryptoK** [comment]
Ok, I have done what you said and found the XHR error. See below

https://bitcoin--casino--no--deposit--bonus-com.cdn.ampproject.org/mbv/s/bitcoin-casino-no-deposit-bonus.com/wp-content/uploads/2022/05/WC-29.mp3?amp_video_host_url=https%3A%2F%2Fbitcoin-casino-no-deposit-bonus.com%2Fweb-stories%2Fbitstarz%2F&amp_video_require_acao_header=1&__amp_source_origin=https%3A%2F%2Fbitcoin-casino-no-deposit-bonus.com 404 

xhr-impl.js

Also the music took more than 5 minutes to kick in on the web story along with 2 pages the voice over is not working. These randomly change from time to time.  When the music did finally kick in the voice over stopped.

**Comment by swissspidy** [comment]
You can safely ignore this XHR error.

Can you confirm that you get JavaScript console errors when you visit the story in incognito mode, yes or no? (See my screenshot above for an example)

**Comment by DanielCryptoK** [comment]
It is strange as the voice over constantly stops working at some point during viewing. Every time I do a check for javascript console errors I get different ones. Below are the current ones showing

Error 1

GET https://rr2---sn-5hoxuj5cax-nwvd.googlevideo.com/videoplayback?expire=1653426542&ei=Ph-NYqC2NIqckAPy0rq4BA&ip=186.176.163.152&id=c7c984caa3b9e15b&itag=742&source=stamp&requiressl=yes&mh=ea&mm=31&mn=sn-5hoxuj5cax-nwvd&ms=au&mv=m&mvi=2&pl=23&susc=stamp&obr=https://bitcoin-casino-no-deposit-bonus.com&acao=yes&mime=video/mp4&vprv=1&gir=yes&clen=601688&dur=17.507&lmt=1634131108748247&mt=1653415262&txp=0011224&sparams=expire,ei,ip,id,itag,source,requiressl,susc,obr,acao,mime,vprv,gir,clen,dur,lmt&sig=AOq0QJ8wRAIgbMvKNSEmgHtG7G1jMn0k45206Xdtl2jHHUVDkD7JPc0CIBgkCqhE00B9N8MQqHc65lEQs2FuYpDgBilR2nwwoCOw&lsparams=mh,mm,mn,ms,mv,mvi,pl&lsig=AG3C_xAwRgIhANcKa0zM2gJ6OFJnl1njmdPGYv1e8wgz9itXxfXzWKCOAiEApv5QMSKMQclkL5-2CN3IJuFHh3nrHNPiQzhQQ_LxPYA= net::ERR_CONNECTION_CLOSED

Error 2

https://bitcoin--casino--no--deposit--bonus-com.cdn.ampproject.org/mbv/s/bitcoin-casino-no-deposit-bonus.com/wp-content/uploads/2022/05/WC-29.mp3?amp_video_host_url=https%3A%2F%2Fbitcoin-casino-no-deposit-bonus.com%2Fweb-stories%2Fbitstarz%2F&amp_video_require_acao_header=1&__amp_source_origin=https%3A%2F%2Fbitcoin-casino-no-deposit-bonus.com 404

https://cdn.ampproject.org/v0.mjs

amp-story-1.0.mjs:2 

Error 3
        
       Uncaught TypeError: Cannot set properties of undefined (setting 'mode')
    at HTMLTrackElement.<anonymous> (amp-story-1.0.mjs:2:40945)

**Comment by DanielCryptoK** [comment]
Yes in incognito, I have just updated another web story with voice over on all pages and subtitles and that gets even stranger errors,  https://bitcoin-casino-no-deposit-bonus.com/web-stories/fairspin/ Music did not start playing and the voice over missing on some pages when tested.

Access to fetch at 'https://bitcoin--casino--no--deposit--bonus-com.cdn.ampproject.org/mbv/s/bitcoin-casino-no-deposit-bonus.com/wp-content/uploads/2022/05/WC-54.mp3?amp_video_host_url=https%3A%2F%2Fbitcoin-casino-no-deposit-bonus.com%2Fweb-stories%2Ffairspin%2F&amp_video_require_acao_header=1&__amp_source_origin=https%3A%2F%2Fbitcoin-casino-no-deposit-bonus.com' from origin 'https://bitcoin-casino-no-deposit-bonus.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

xhr-impl.js:75 
             
        
       GET https://bitcoin--casino--no--deposit--bonus-com.cdn.ampproject.org/mbv/s/bitcoin-casino-no-deposit-bonus.com/wp-content/uploads/2022/05/WC-54.mp3?amp_video_host_url=https%3A%2F%2Fbitcoin-casino-no-deposit-bonus.com%2Fweb-stories%2Ffairspin%2F&amp_video_require_acao_header=1&__amp_source_origin=https%3A%2F%2Fbitcoin-casino-no-deposit-bonus.com net::ERR_FAILED 500

GET https://rr1---sn-5hoxuj5cax-nwvd.googlevideo.com/videoplayback?expire=1653429962&ei=miyNYouZGuq5j-8Pr_WqsAg&ip=186.176.163.152&id=02cd999892bb1a4b&itag=18&source=stamp&requiressl=yes&mh=yP&mm=31&mn=sn-5hoxuj5cax-nwvd&ms=au&mv=m&mvi=1&pl=23&susc=stamp&obr=https://bitcoin-casino-no-deposit-bonus.com&acao=yes&mime=video/mp4&vprv=1&dur=10.066&lmt=1653317805226490&mt=1653419110&txp=0011224&sparams=expire,ei,ip,id,itag,source,requiressl,susc,obr,acao,mime,vprv,dur,lmt&sig=AOq0QJ8wRAIgEplKgWgCShNSo5tT2tKRXwXv_pYQbfWGDsQq05bmlkgCIFPnspURBGcFAYimEMlz5wty6zOL-2qUZfraLeHtXo8g&lsparams=mh,mm,mn,ms,mv,mvi,pl&lsig=AG3C_xAwRQIhAO-MSZi3RFNalBH6bQhAfW7IKSbVR7Gc6VG51nBwlSrCAiBcFhCboyKR6uxhToRNEoJIO5b-TskRJ2XPy9O4Nf-V1g== net::ERR_CONNECTION_CLOSED

**Comment by swissspidy** [comment]
Can you try disabling Video Cache in the Web Stories settings and try again?

**Comment by DanielCryptoK** [comment]
Yes, Finally that worked to get rid of all the errors on both of the web stories :)

Please can you tell me how to sort out this warning from happening, as I dont know what is being used to preload the link

The resource https://cdn.ampproject.org/v0/amp-story-1.0.js was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

The resource https://cdn.ampproject.org/v0/amp-story-1.0.js was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

**Comment by swissspidy** [comment]
> Please can you tell me how to sort out this warning from happening, as I dont know what is being used to preload the link

You can ignore this warning. There is nothing on your side that you can do to get rid of it. But it does not harm in any way.

We'll take a look into somehow hiding it.

**Comment by DanielCryptoK** [comment]
All of the same errors though are showing on the AMP test?? Does that mean it is still broken?

**Comment by swissspidy** [comment]
At this point I think it would be best to open an issue at https://github.com/ampproject/amphtml as these error messages now really seem to be coming from AMP itself.

**Comment by DanielCryptoK** [comment]
I am going to delete AMP tomorrow as it is no good these days and even some brewers don't accept it I have heard. Hopefully once all amp plugins apart from web stories is done and the amp pages redirected to non amp, it should fix the problem but I dont think it will as my other site does not have AMP installed and the web stories for that one get loads of errors but different ones than this one. Its crazy it been well over 1 year and not once have I had a web story displaying without an error, which probably explained why I get an average of 5 views per day from 150 web stories !!!
